src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -54,7 +53,7 @@ public WebClientDownloader(string userName, string password, ILogger logger, str
 
             if (securityProtocolHandler.SecurityProtocol != SystemDefault)
             {
-                securityProtocolHandler.SecurityProtocol = Tls13 | SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
+                securityProtocolHandler.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
```

Comment:
I don't get this. In case we run under .NET Fx 4.8-only system, the Tls13 value eshould work. (I'm not reviewing, just curious)

Reply:
For .NET Core 2.0 and .NET Framework 4.7 and upwards the SystemDefault value exists and that will be the default value, and we will not set the property. This means that the enabled protocols will be decided by the OS.
For .NET Framework versions below 4.7 we will set the property and TLS 1.2, 1.1 and 1.0 will be enabled.

As the TLS 1.2 support starts at Net Framework 4.5 this means we don't support .NET Framework versions below 4.5 (This was the previous behavior too.)
As for TLS 1.3 support, it will be enabled only for 4.8 with the new implementation as that is the only one supporting it and for .NET Core it will be from version 3.0.

Reply:
What is the default value of `SecurityProtocol` on .NET Fx 4.5, 4.6 and 4.7?

Reply:
4.5: Ssl3 and Tls 1.0
4.6: Tls 1.0, 1.1 and 1.2
4.7: SystemDefault

Reply:
the main idea here is:
- if it has system default, leave it - like this we ensure that we won't lower the security by overwriting the value
- otherwise, keep the old S4NET behavior

Reply:
Additionally we not just not "lower" the security but also move the decision to which protocols to enable to the users.
SystemDefault value means that the protocols enabled by the OS will be used. (That means whatever the user configures for their OS will be used)

Reply:
Please do intensive testing on machines with only those environments present. I once spent some time trying to understand this issue and I think it doesn't behave reasonably on different versions.

As a reference, what it really does deep down can be found here for .NET Fx 4.8: (https://referencesource.microsoft.com/#System/net/System/Net/_SecureChannel.cs,119537a23fb639d4) 
This piece varies a lot for different Fx versions and doesn't always makes sense.

Reply:
My concern is that the behavior of `SystemDefault` enum value is not what we expect it to be on different frameworks. As it was hardcoded (not configurable by system actually) and stuck to some old config on some of those versions.

Reply:
You're right @pavel-mikula-sonarsource .

For the bugfix we want it not to fail.